### PR TITLE
Typescript

### PR DESCRIPTION
**Pull request by drzax**
Convert to typescript. This is step one, it has the same API so no breaking changes. However, this process has brought up a few issues to consider:

* This has been used in the past with [this typing](https://github.com/abcnews/scrollyteller/blob/master/src/types.ts#L1) which is a nice idea, but in reality the code can't currently guarantee that array values in the returned object will have a consistent type so the types defined here don't currently match how it's been used (and therefore, in typescript land, this might be a breaking change)
* Maybe we could make it guarantee consistent types within returned arrays, but that would also be a breaking change
* We should consider allowing consumers to pass in their own types to define what they expect to get back.

### CODE REVIEW

**Comment by drzax** [comment]
That CI failure is because it's also trying to build master to check the difference in package size. It's all overkill, but cool to have once this is merged so I'll leave it.

**Review by colingourlay — APPROVED** [review]

**Review by phocks — APPROVED** [review]
Seems good to me

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      value = null;
+    }
+
+    let propVal = o[prop];
+
+    if (propVal !== undefined) {
+      // Prop exists so assume it should be an array
+      if (!Array.isArray(propVal)) {
+        propVal = [propVal];
+      }
```
Comment (colingourlay) [inline src/index.ts:92]: @drzax we could compare types after this (`typeof value === typeof propVal[0]`) and throw if they're different. TypeScript _should_ be smart enough to allow you to push after that gate is passed

Comment (drzax) [reply src/index.ts:92]: Hahaha! If only it were that easy (see 1cb9943a76717e215151210d1e0854185dd9b2d9).

**Comment by drzax** [comment]
I've updated this somewhat with the intention of matching expectations of the only [typescript use of the library I'm aware](https://github.com/abcnews/scrollyteller/blob/v3.0.1/src/types.ts). That's not how scrollyteller will work now anyway, but I think it's a better typing to be able to guarantee type consistency within returned arrays. This is probably still a breaking change. Thoughts?

**Comment by drzax** [comment]
Now that I've let this sit for 5 months, I re-read the code and didn't hate it, so I'm going to merge today and release a major bump if there are no objections.

**Review by phocks — APPROVED** [review]
Seems good to me!

**Pull request merged**